dipy/segment/bundles.py

```diff
@@ -335,15 +335,20 @@ def recognize(self, model_bundle, model_clust_thr,
         Parameters
         ----------
         model_bundle : Streamlines
+            model bundle streamlines used as a reference to extract similar
+            streamlines from input tractogram
         model_clust_thr : float
+            MDF distance threshold for the model bundles
         reduction_thr : float
+            Reduce search space in the target tractogram by (mm) (default 10)
         reduction_distance : string
-            mdf or mam (default mdf)
+            Reduction distance type can be mdf or mam (default mdf)
         slr : bool
             Use Streamline-based Linear Registration (SLR) locally
             (default True)
         slr_metric : BundleMinDistanceMetric
         slr_x0 : array
+            Transformation allowed. translation, rigid, similarity or scaling
```

Comment:
Does this have to be a 4x4 array?

Reply:
Hi @arokem ,

Datatype for `slr_x0` should be a `string` instead of an array. This takes a single string value either translation, rigid, similarity, or scaling.

I will update it.

Reply:
I just realized `slr_x0 `could be an `array or int or str`

Similar to what we have in SLR registration:

            Initial parametrization for the optimization.

            If 1D array with:
                a) 6 elements then only rigid registration is performed with
                the 3 first elements for translation and 3 for rotation.
                b) 7 elements also isotropic scaling is performed (similarity).
                c) 12 elements then translation, rotation (in degrees),
                scaling and shearing are performed (affine).

                Here is an example of x0 with 12 elements:
                ``x0=np.array([0, 10, 0, 40, 0, 0, 2., 1.5, 1, 0.1, -0.5, 0])``

                This has translation (0, 10, 0), rotation (40, 0, 0) in
                degrees, scaling (2., 1.5, 1) and shearing (0.1, -0.5, 0).

            If int:
                a) 6
                    ``x0 = np.array([0, 0, 0, 0, 0, 0])``
                b) 7
                    ``x0 = np.array([0, 0, 0, 0, 0, 0, 1.])``
                c) 12
                    ``x0 = np.array([0, 0, 0, 0, 0, 0, 1., 1., 1, 0, 0, 0])``

            If str:
                a) "rigid"
                    ``x0 = np.array([0, 0, 0, 0, 0, 0])``
                b) "similarity"
                    ``x0 = np.array([0, 0, 0, 0, 0, 0, 1.])``
                c) "affine"
                    ``x0 = np.array([0, 0, 0, 0, 0, 0, 1., 1., 1, 0, 0, 0])